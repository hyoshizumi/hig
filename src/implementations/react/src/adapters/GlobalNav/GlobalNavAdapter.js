import * as HIG from 'hig-vanilla';
import * as PropTypes from 'prop-types';

import HIGElement from '../../elements/HIGElement';
import HIGChildValidator from '../../elements/HIGChildValidator';
import createComponent from '../createComponent';

import SideNavAdapterComponent, { SideNavAdapter } from './SideNav/SideNavAdapter';
import TopNavAdapterComponent, {
  TopNavAdapter
} from './TopNav/TopNavAdapter';
import SubNavAdapterComponent, {
  SubNavAdapter
} from './SubNav/SubNavAdapter';
import Slot from '../../elements/components/GlobalNav/Slot';
import SideNav from '../../elements/components/GlobalNav/SideNav';

class GlobalNavAdapter extends HIGElement {
  constructor(initialProps) {
    super(HIG.GlobalNav, initialProps);
  }

  componentDidMount() {
    // Add any children
    if (this.sideNav) {
      this.hig.addSideNav(this.sideNav.hig);
      this.sideNav.mount();
    }

    if (this.topNav) {
      this.hig.addTopNav(this.topNav.hig);
      this.topNav.mount();
    }

    if (this.subNav) {
      this.hig.addSubNav(this.subNav.hig);
      this.subNav.mount();
    }

    if (this.initialProps.sideNavOpen) {
      this.hig.showSideNav();
    } else {
      this.hig.hideSideNav();
    }

    if (this.slot) {
      this.hig.addSlot(this.slot);
    }
  }

  createElement(ElementConstructor, props) {
    switch (ElementConstructor) {
      case TopNavAdapter:
        return new TopNavAdapter(this.hig.partials.TopNav, props);
      case SubNavAdapter:
        return new SubNavAdapter(this.hig.partials.SubNav, props);
      case SideNavAdapter:
        return new SideNavAdapter(this.hig.partials.SideNav, props);
      default:
        throw new Error(`Unknown type ${ElementConstructor.name}`);
    }
  }

  appendChild(instance, beforeChild = {}) {
    if (instance instanceof SideNavAdapter) {
      if (this.sideNav) {
        throw new Error('only one SideNav is allowed');
      } else {
        this.sideNav = instance;
        if (this.mounted) {
          this.hig.addSideNav(instance.hig);
          instance.mount();
        }
      }
    } else if (instance instanceof TopNavAdapter) {
      if (this.topNav) {
        throw new Error('only one TopNav is allowed');
      } else {
        this.topNav = instance;
        if (this.mounted) {
          this.hig.addTopNav(instance.hig);
          instance.mount();
        }
      }
    } else if (instance instanceof SubNavAdapter) {
      if (this.subNav) {
        throw new Error('only one SubNav is allowed');
      } else {
        this.subNav = instance;
        if (this.mounted) {
          this.hig.addSubNav(instance.hig);
          instance.mount();
        }
      }
    } else {
      throw new Error('unknown type');
    }
  }

  addSlot(element) {
    if (this.mounted) {
      this.hig.addSlot(element);
    } else {
      this.slot = element;
    }
  }

  insertBefore(instance) {
    this.appendChild(instance);
  }

  removeChild(instance) {
    if (instance instanceof SideNavAdapter) {
      this.sideNav = null;
    } else if (instance instanceof TopNavAdapter) {
      this.topNav = null;
    } else if (instance instanceof SubNavAdapter) {
      this.subNav = null;
    }
    instance.unmount();
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case 'sideNavOpen': {
          propValue ? this.hig.showSideNav() : this.hig.hideSideNav();
          break;
        }
        case 'children': {
          // No-op
          break;
        }
        default: {
          console.warn(`${propKey} is unknown`);
        }
      }
    }
  }
}

const GlobalNavAdapterComponent = createComponent(GlobalNavAdapter);

GlobalNavAdapterComponent.propTypes = {
  sideNavOpen: PropTypes.bool,
  children: HIGChildValidator([
    SideNav,
    SideNavAdapterComponent,
    TopNavAdapterComponent,
    SubNavAdapterComponent,
    Slot,
  ])
};

GlobalNavAdapterComponent.__docgenInfo = {
  props: {
    sideNavOpen: {
      description: 'show or hide the SideNav'
    },

    children: {
      description: 'support adding SideNav, SubNav, or TopNav'
    }
  }
};

GlobalNavAdapterComponent.SideNav = SideNavAdapterComponent;
GlobalNavAdapterComponent.TopNav = TopNavAdapterComponent;
GlobalNavAdapterComponent.SubNav = SubNavAdapterComponent;
GlobalNavAdapterComponent.Slot = Slot;

export default GlobalNavAdapterComponent;
