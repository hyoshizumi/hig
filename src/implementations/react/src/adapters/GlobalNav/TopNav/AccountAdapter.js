import * as PropTypes from 'prop-types';
import createComponent from '../../createComponent';
import HIGElement from '../../../elements/HIGElement';

export class AccountAdapter extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);

    this.props = { ...initialProps };
  }

  componentDidMount() {
    if (this.initialProps.active) {
      this.hig.activate();
    } else {
      this.hig.deactivate();
    }
  }

  commitUpdate(updatePayload) {
    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case 'active': {
          if (propValue) {
            this.hig.activate();
          } else {
            this.hig.deactivate();
          }
          break;
        }
        case 'image': {
          this.hig.setImage(propValue);
          break;
        }
        case 'label': {
          this.hig.setLabel(propValue);
          break;
        }
        case 'onClick': {
          const dispose = this._disposeFunctions.get('onClickDispose');

          if (dispose) {
            dispose();
          }

          this._disposeFunctions.set(
            'onClickDispose',
            this.hig.onClick(propValue)
          );
          break;
        }
        default: {
          // no-op
        }
      }
    }
  }
}

const AccountComponent = createComponent(AccountAdapter);

AccountComponent.propTypes = {
  image: PropTypes.string,
  label: PropTypes.string,
  active: PropTypes.bool,
  onClick: PropTypes.func
};

AccountComponent.__docgenInfo = {
  props: {
    label: {
      description: 'sets {String} the label displayed for an account in Account/account switcher'
    },
    image: {
      description: 'sets {String} the image displayed for an account in Account/account switcher'
    },
    active: {
      description: '{func} activates the Account'
    },
    onClick: {
      description: '{func} calls the provided callback when user clicks on the Account'
    }
  }
};

export default AccountComponent;
