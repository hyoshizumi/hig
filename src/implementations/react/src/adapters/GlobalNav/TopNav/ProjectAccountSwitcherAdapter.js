import * as PropTypes from "prop-types";
import HIGElement from "../../../elements/HIGElement";
import HIGNodeList from "../../../elements/HIGNodeList";
import HIGChildValidator from "../../../elements/HIGChildValidator";
import createComponent from "../../createComponent";
// import AccountComponent, { AccountAdapter } from "./AccountAdapter";
// import ProjectComponent, { ProjectAdapter } from "./ProjectAdapter";
// import Project from "../../../elements/components/GlobalNav/TopNav/Project";
// import Account from "../../../elements/components/GlobalNav/TopNav/Account";

import ListComponent, { ListAdapter } from "./ListAdapter";

export class ProjectAccountSwitcherAdapter extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);

    this.lists = new HIGNodeList({
      ListAdapter: {
        type: ListAdapter,
        HIGConstructor: this.hig.partials.List,
        onAdd: (instance, beforeInstance) => {
          this.hig.addList(instance, beforeInstance);
        }
      }
    });

    
  }

  componentDidMount() {
    this.lists.componentDidMount();

    if (this.initialProps.open === true) {
      this.hig.open();
    }
    if (this.initialProps.showCaret) {
      this.hig.showCaret();
    } else {
      this.hig.hideCaret();
    }

    this.commitUpdate(this.props);
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case "open": {
          if (propValue) {
            this.hig.open();
          } else {
            this.hig.close();
          }
          break;
        }
        case "showCaret": {
          if (propValue) {
            this.hig.showCaret();
          } else {
            this.hig.hideCaret();
          }
          break;
        }
        case "activeLabel": {
          this.hig.setActiveLabel(propValue);
          break;
        }
        case "activeType": {
          this.hig.setActiveType(propValue);
          break;
        }
        case "activeImage": {
          this.hig.setActiveImage(propValue);
          break;
        }
        case "onClickOutside": {
          const dispose = this._disposeFunctions.get("onClickOutsideDispose");

          if (dispose) {
            dispose();
          }

          this._disposeFunctions.set(
            "onClickOutisdeDispose",
            this.hig.onClickOutside(propValue)
          );
          break;
        }
        case "onClick": {
          const dispose = this._disposeFunctions.get("onClick");

          if (dispose) {
            dispose();
          }

          this._disposeFunctions.set("onClick", this.hig.onClick(propValue));
          break;
        }
        case "children": {
          //no-op
          break;
        }
        default: {
          console.warn(`${propKey} is unknown`);
        }
      }
    }
  }

  createElement(ElementConstructor, props) {
    return this.lists.createElement(ElementConstructor, props);
  }

  insertBefore(instance, beforeChild = {}) {
    this.lists.insertBefore(instance, beforeChild);
  }
}

const ProjectAccountSwitcherComponent = createComponent(
  ProjectAccountSwitcherAdapter
);

ProjectAccountSwitcherComponent.propTypes = {
  open: PropTypes.bool,
  showCaret: PropTypes.bool,
  onAccountChange: PropTypes.func,
  onProjectChange: PropTypes.func,
  setProjectSectionTitle: PropTypes.func,
  setAccountSectionTitle: PropTypes.func,
  activeLabel: PropTypes.string,
  activeImage: PropTypes.string,
  activeType: PropTypes.string,
  children: HIGChildValidator([
    ListAdapter
  ])
};

ProjectAccountSwitcherComponent.__docgenInfo = {
  props: {
    open: {
      description: "{bool} opens the project/account switcher"
    },
    showCaret: {
      description: "shows a caret indicating a flyout in Project Account Switcher"
    },
    activeLabel: {
      description: "Sets the label displayed in the top nav"
    },
    activeImage: {
      description: "Sets the image displayed in the top nav"
    },
    activeType: {
      description: "Sets the type of the item displayed in the top nav"
    },
    onClickOutside: {
      description: "Calls the provided callback when the switcher is open and the user clicks outside the switcher"
    },
    onClick: {
      description: "Calls the provided callback when user clicks on the switcher in the top nav"
    },
    children: {
      description: "support adding Project and Account"
    }
  }
};

export default ProjectAccountSwitcherComponent;
