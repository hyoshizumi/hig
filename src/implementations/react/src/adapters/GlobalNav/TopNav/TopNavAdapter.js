import * as PropTypes from 'prop-types';

import HIGElement from '../../../elements/HIGElement';
import HIGChildValidator from '../../../elements/HIGChildValidator';
import createComponent from '../../createComponent';

import HIGNodeList from '../../../elements/HIGNodeList';
import ProjectAccountSwitcherComponent, {
  ProjectAccountSwitcherAdapter
} from './ProjectAccountSwitcherAdapter';
import ProjectAccountSwitcher
  from '../../../elements/components/GlobalNav/TopNav/ProjectAccountSwitcher';
import ProfileComponent, {
  ProfileAdapter
} from './ProfileAdapter';
import ShortcutAdapterComponent, { ShortcutAdapter } from './ShortcutAdapter';
import HelpComponent, { HelpAdapter } from './HelpAdapter';
import SearchComponent, {
  SearchAdapter
} from './SearchAdapter';

export class TopNavAdapter extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);

    this.shortcuts = new HIGNodeList({
      ShortcutAdapter: {
        type: ShortcutAdapter,
        HIGConstructor: this.hig.partials.Shortcut,
        onAdd: (instance, beforeInstance) => {
          this.hig.addShortcut(instance, beforeInstance);
        }
      }
    });
  }

  componentDidMount() {
    // Add any children
    if (this.profile) {
      this.hig.addProfile(this.profile.hig);
      this.profile.mount();
    }

    if (this.projectAccountSwitcher) {
      this.hig.addProjectAccountSwitcher(this.projectAccountSwitcher.hig);
      this.projectAccountSwitcher.mount();
    }

    this.shortcuts.componentDidMount();

    if (this.help) {
      this.hig.addHelp(this.help.hig);
      this.help.mount();
    }

    if (this.search) {
      this.hig.addSearch(this.search.hig);
      this.search.mount();
    }
  }

  commitUpdate(updatePayload) {
    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case 'children': {
          // No-op
          break;
        }
        case 'logo': {
          this.hig.setLogo(propValue);
          break;
        }
        case 'logoLink': {
          this.hig.setLogoLink(propValue);
          break;
        }
        case 'onLogoClick': {
          const dispose = this._disposeFunctions.get('onLogoClickDispose');

          if (dispose) {
            dispose();
          }

          this._disposeFunctions.set(
            'onLogoClickDispose',
            this.hig.onLogoClick(propValue)
          );
          break;
        }
        default: {
          // No-op
        }
      }
    }
  }

  createElement(ElementConstructor, props) {
    switch (ElementConstructor) {
      case ProfileAdapter:
        return new ProfileAdapter(this.hig.partials.Profile, props);
      case ProjectAccountSwitcherAdapter:
        return new ProjectAccountSwitcherAdapter(
          this.hig.partials.ProjectAccountSwitcher,
          props
        );
      case HelpAdapter:
        return new HelpAdapter(this.hig.partials.Help, props);
      case ShortcutAdapter:
        return this.shortcuts.createElement(ElementConstructor, props);
      case SearchAdapter:
        return new SearchAdapter(this.hig.partials.Search, props);
      default:
        throw new Error(`Unknown type ${ElementConstructor.name}`);
    }
  }

  appendChild(instance) {
    this.requireSingleInstance(instance);
    this.checkValidChild(instance);

    if (instance instanceof ShortcutAdapter) {
      this.shortcuts.insertBefore(instance);
    } else {
      this[this.getPropertyNameFor(instance)] = instance;
      if (this.mounted) {
        this.hig[this.getFunctionNameFor(instance)](instance.hig);
        instance.mount();
      }
    }
  }

  insertBefore(instance) {
    this.appendChild(instance);
  }

  removeChild(instance) {
    if (instance instanceof ProfileAdapter) {
      this.profile = null;
    }

    instance.unmount();
  }

  requireSingleInstance(instance) {
    const requiredSingle = [
      'Profile',
      'ProjectAccountSwitcher',
      'HelpAdapter',
      'SearchAdapter'
    ];
    super.requireSingleInstance(instance, requiredSingle);
  }

  checkValidChild(instance) {
    const validInstances = [
      ProfileAdapter,
      ProjectAccountSwitcherAdapter,
      ShortcutAdapter,
      HelpAdapter,
      SearchAdapter
    ];
    if (!validInstances.includes(instance.constructor)) {
      throw new Error(
        `${instance.constructor.name} is not a valid child element of this parent.`
      );
    }
  }

  getPropertyNameFor(instance) {
    if (instance instanceof ProfileAdapter) {
      return 'profile';
    }
    if (instance instanceof ProjectAccountSwitcherAdapter) {
      return 'projectAccountSwitcher';
    }
    if (instance instanceof ShortcutAdapter) {
      return 'shortcut';
    }
    if (instance instanceof HelpAdapter) {
      return 'help';
    }
    if (instance instanceof SearchAdapter) {
      return 'search';
    }
    return null;
  }

  getFunctionNameFor(instance) {
    return `add${instance.constructor.name}`;
  }

  onHamburgerClick(callback) {
    this.hig.onHamburgerClick(callback);
  }
}

const TopNavAdapterComponent = createComponent(TopNavAdapter);

TopNavAdapterComponent.propTypes = {
  logo: PropTypes.string,
  logoLink: PropTypes.string,
  onLogoClick: PropTypes.func,
  onHamburgerClick: PropTypes.func,
  children: HIGChildValidator([
    ProfileComponent,
    ProjectAccountSwitcherComponent,
    ProjectAccountSwitcher,
    ShortcutAdapterComponent,
    HelpComponent,
    SearchComponent
  ])
};

TopNavAdapterComponent.__docgenInfo = {
  props: {
    logo: {
      description: 'sets the logo'
    },
    logoLink: {
      description: 'sets the link of the logo'
    },
    onHamburgerClick: {
      description: 'triggered when hamburger icon is clicked'
    },
    onLogoClick: {
      description: 'triggered when logo is clicked'
    }
  }
};

TopNavAdapterComponent.Profile = ProfileComponent;
TopNavAdapterComponent.Shortcut = ShortcutAdapterComponent;
TopNavAdapterComponent.Help = HelpComponent;
TopNavAdapterComponent.ProjectAccountSwitcher = ProjectAccountSwitcherComponent;
TopNavAdapterComponent.Search = SearchComponent;

export default TopNavAdapterComponent;
