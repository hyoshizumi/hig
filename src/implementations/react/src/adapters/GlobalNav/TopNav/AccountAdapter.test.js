import { mount } from 'enzyme';
import * as HIG from 'hig-vanilla';
import React from 'react';

import GlobalNavAdapter from '../GlobalNavAdapter';
import TopNavAdapter from './TopNavAdapter';
import ProjectAccountSwitcherAdapter from './ProjectAccountSwitcherAdapter';
import AccountAdapter from './AccountAdapter';

const Context = props => (
  <GlobalNavAdapter>
    <TopNavAdapter>
      <ProjectAccountSwitcherAdapter>
        <AccountAdapter {...props} />
      </ProjectAccountSwitcherAdapter>
    </TopNavAdapter>
  </GlobalNavAdapter>
);

function createOrionAccount(props) {
  const reactContainer = document.createElement('div');
  const reactWrapper = mount(<Context {...props} />, {
    attachTo: reactContainer
  });

  return { reactWrapper, reactContainer };
}

function createHigContext(props) {
  const higContainer = document.createElement('div');

  const higNav = new HIG.GlobalNav();
  higNav.mount(higContainer);

  const higTopNav = new higNav.partials.TopNav({});
  higNav.addTopNav(higTopNav);

  const projectAccountSwitcher = new higTopNav.partials.ProjectAccountSwitcher({
  });

  higTopNav.addProjectAccountSwitcher(projectAccountSwitcher);
  projectAccountSwitcher.hideCaret();

  const account = new projectAccountSwitcher.partials.Account(props);
  projectAccountSwitcher.addAccount(account);

  return { higContainer, higItem: account };
}

describe('<Account>', () => {
  it('renders an Account', () => {
    const defaults = {};
    const { higContainer } = createHigContext(defaults);
    const { reactContainer } = createOrionAccount(defaults);

    expect(reactContainer.firstElementChild.outerHTML).toMatchSnapshot();

    expect(reactContainer.firstElementChild.outerHTML).toEqual(
      higContainer.firstElementChild.outerHTML
    );
  });

  it('setting and updating props', () => {
    const defaults = {
      image: 'test.jpg',
      label: 'testLabel'
    };

    const nextProps = {
      image: 'newImage.jpg',
      label: 'newLabel',
      active: true
    };

    const { higContainer, higItem } = createHigContext(defaults);
    const { reactWrapper, reactContainer } = createOrionAccount(defaults);

    reactWrapper.setProps(nextProps);

    higItem.setImage(nextProps.image);
    higItem.setLabel(nextProps.label);
    higItem.activate();

    expect(reactContainer.firstElementChild.outerHTML).toMatchSnapshot();
    expect(reactContainer.firstElementChild.outerHTML).toEqual(
      higContainer.firstElementChild.outerHTML
    );
  });

  it('warns when passed an unsupported property', () => {
    const warnSpy = jest.fn();
    const { reactWrapper } = createOrionAccount({});
    console.warn = warnSpy;

    reactWrapper.setProps({ realProp: false });

    expect(warnSpy).toHaveBeenCalled();
  });

  ['onClick'].forEach((eventName) => {
    it(`handles ${eventName}`, () => {
      const warnSpy = jest.fn();
      const { reactWrapper } = createOrionAccount({});
      console.warn = warnSpy;

      reactWrapper.setProps({ [eventName]: () => {} });

      expect(warnSpy).not.toHaveBeenCalled();
    });
  });
});
