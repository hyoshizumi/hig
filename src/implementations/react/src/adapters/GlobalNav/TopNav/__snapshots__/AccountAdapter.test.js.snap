// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Account> renders an Account 1`] = `
"<div class=\\"hig__global-nav\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <div class=\\"hig__global-nav__top-nav\\">
  <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--left\\">
    <div class=\\"hig__global-nav__top-nav__hamburger\\">
      <div class=\\"hig__global-nav__top-nav__hamburger__hamburgericon\\"><div class=\\"hig__icon\\"><svg width=\\"16\\" height=\\"12\\" viewBox=\\"0 0 16 12\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"hamburger\\"><g transform=\\"translate(-4 -6)\\" fill=\\"#353636\\" fill-rule=\\"evenodd\\"><rect x=\\"4\\" y=\\"6\\" width=\\"16\\" height=\\"2\\"></rect><rect x=\\"4\\" y=\\"11\\" width=\\"16\\" height=\\"2\\"></rect><rect x=\\"4\\" y=\\"16\\" width=\\"16\\" height=\\"2\\"></rect></g></svg></div></div>
      <div class=\\"hig__global-nav__top-nav__hamburger__closeicon\\"><div class=\\"hig__icon\\"><svg width=\\"18\\" height=\\"18\\" viewBox=\\"0 0 18 18\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-hamburger\\"><g stroke=\\"#0A131C\\" stroke-width=\\"2\\" fill=\\"none\\" fill-rule=\\"evenodd\\" stroke-linecap=\\"square\\"><path d=\\"M18.754628,5.24537196 L5.24537196,18.754628\\" transform=\\"translate(-3 -3)\\"></path><path d=\\"M5.24537196,5.24537196 L18.754628,18.754628\\" transform=\\"translate(-3 -3)\\"></path></g></svg></div></div>
    </div>
    <div class=\\"hig__global-nav__top-nav__logo\\">
      <a>
        <img src=\\"\\">
      </a>
    </div>
  </div>
  <!--SEARCH-->
  <div class=\\"hig__global-nav__top-nav__spacer\\"></div>
  <div class=\\"hig__global-nav__top-nav____project-account-switcher\\">
    <div class=\\"hig__flyout hig__flyout--anchor-top-right\\">
    <div class=\\"hig__global-nav__top-nav__project-account-switcher__target\\">
    <div class=\\"hig__global-nav__top-nav__project-account-switcher__item hig__global-nav__top-nav__project-account-switcher__item--project\\">
    <span class=\\"hig__global-nav__top-nav__project-account-switcher__item__image-wrapper\\">
        <img class=\\"hig__global-nav__top-nav__project-account-switcher__item__image\\" src=\\"\\" alt=\\"\\" width=\\"32\\" height=\\"32\\">
        <span class=\\"hig__global-nav__top-nav__project-account-switcher__item__image-placeholder\\"></span>
    </span>
    <span class=\\"hig__global-nav__top-nav__project-account-switcher__item__label\\"></span>
</div><!--ITEM-->
    <span class=\\"hig__global-nav__top-nav__project-account-switcher__target__caret hig__global-nav__top-nav__project-account-switcher__target__caret--hide\\"><svg width=\\"10\\" height=\\"6\\" viewBox=\\"0 0 10 6\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"caret\\"><title>hig-web-light/basic/form-element/caret</title><polyline transform=\\"rotate(45 9.828 3)\\" points=\\"8 2 8 8 2 8\\" stroke=\\"#2E4258\\" stroke-width=\\"1.15\\" fill=\\"none\\" fill-rule=\\"evenodd\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></polyline></svg></span>
</div><!--TARGET-->
    <div class=\\"hig__flyout__container\\">
        <div class=\\"hig__flyout__chevron\\"></div>
        <div class=\\"hig__flyout__panel\\">
            <div class=\\"hig__global-nav__top-nav__project-account-switcher__lists\\">
    <div class=\\"hig__global-nav__top-nav__project-account-switcher__lists__list\\">
        <div class=\\"hig__global-nav__top-nav__project-account-switcher__item hig__global-nav__top-nav__project-account-switcher__item--account\\">
    <span class=\\"hig__global-nav__top-nav__project-account-switcher__item__image-wrapper\\">
        <img class=\\"hig__global-nav__top-nav__project-account-switcher__item__image\\" src=\\"\\" alt=\\"\\" width=\\"32\\" height=\\"32\\">
        <span class=\\"hig__global-nav__top-nav__project-account-switcher__item__image-placeholder\\"></span>
    </span>
    <span class=\\"hig__global-nav__top-nav__project-account-switcher__item__label\\"></span>
</div><!--ACCOUNTS-->
    </div>
    <div class=\\"hig__global-nav__top-nav__project-account-switcher__lists__list\\">
        <!--PROJECTS-->
    </div>
</div><!--SLOT-->
        </div>
    </div>
</div><!--FLYOUT-->
</div><!--PROJECT_ACCOUNT_SWITCHER-->
  <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--right\\">
  <!--SHORTCUT-->
  </div>
  <!--PROFILE-->
</div><!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<Account> setting and updating props 1`] = `
"<div class=\\"hig__global-nav\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <div class=\\"hig__global-nav__top-nav\\">
  <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--left\\">
    <div class=\\"hig__global-nav__top-nav__hamburger\\">
      <div class=\\"hig__global-nav__top-nav__hamburger__hamburgericon\\"><div class=\\"hig__icon\\"><svg width=\\"16\\" height=\\"12\\" viewBox=\\"0 0 16 12\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"hamburger\\"><g transform=\\"translate(-4 -6)\\" fill=\\"#353636\\" fill-rule=\\"evenodd\\"><rect x=\\"4\\" y=\\"6\\" width=\\"16\\" height=\\"2\\"></rect><rect x=\\"4\\" y=\\"11\\" width=\\"16\\" height=\\"2\\"></rect><rect x=\\"4\\" y=\\"16\\" width=\\"16\\" height=\\"2\\"></rect></g></svg></div></div>
      <div class=\\"hig__global-nav__top-nav__hamburger__closeicon\\"><div class=\\"hig__icon\\"><svg width=\\"18\\" height=\\"18\\" viewBox=\\"0 0 18 18\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-hamburger\\"><g stroke=\\"#0A131C\\" stroke-width=\\"2\\" fill=\\"none\\" fill-rule=\\"evenodd\\" stroke-linecap=\\"square\\"><path d=\\"M18.754628,5.24537196 L5.24537196,18.754628\\" transform=\\"translate(-3 -3)\\"></path><path d=\\"M5.24537196,5.24537196 L18.754628,18.754628\\" transform=\\"translate(-3 -3)\\"></path></g></svg></div></div>
    </div>
    <div class=\\"hig__global-nav__top-nav__logo\\">
      <a>
        <img src=\\"\\">
      </a>
    </div>
  </div>
  <!--SEARCH-->
  <div class=\\"hig__global-nav__top-nav__spacer\\"></div>
  <div class=\\"hig__global-nav__top-nav____project-account-switcher\\">
    <div class=\\"hig__flyout hig__flyout--anchor-top-right\\">
    <div class=\\"hig__global-nav__top-nav__project-account-switcher__target\\">
    <div class=\\"hig__global-nav__top-nav__project-account-switcher__item hig__global-nav__top-nav__project-account-switcher__item--project\\">
    <span class=\\"hig__global-nav__top-nav__project-account-switcher__item__image-wrapper\\">
        <img class=\\"hig__global-nav__top-nav__project-account-switcher__item__image\\" src=\\"\\" alt=\\"\\" width=\\"32\\" height=\\"32\\">
        <span class=\\"hig__global-nav__top-nav__project-account-switcher__item__image-placeholder\\"></span>
    </span>
    <span class=\\"hig__global-nav__top-nav__project-account-switcher__item__label\\"></span>
</div><!--ITEM-->
    <span class=\\"hig__global-nav__top-nav__project-account-switcher__target__caret hig__global-nav__top-nav__project-account-switcher__target__caret--hide\\"><svg width=\\"10\\" height=\\"6\\" viewBox=\\"0 0 10 6\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"caret\\"><title>hig-web-light/basic/form-element/caret</title><polyline transform=\\"rotate(45 9.828 3)\\" points=\\"8 2 8 8 2 8\\" stroke=\\"#2E4258\\" stroke-width=\\"1.15\\" fill=\\"none\\" fill-rule=\\"evenodd\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></polyline></svg></span>
</div><!--TARGET-->
    <div class=\\"hig__flyout__container\\">
        <div class=\\"hig__flyout__chevron\\"></div>
        <div class=\\"hig__flyout__panel\\">
            <div class=\\"hig__global-nav__top-nav__project-account-switcher__lists\\">
    <div class=\\"hig__global-nav__top-nav__project-account-switcher__lists__list\\">
        <div class=\\"hig__global-nav__top-nav__project-account-switcher__item hig__global-nav__top-nav__project-account-switcher__item--account hig__global-nav__top-nav__project-account-switcher__item--active\\">
    <span class=\\"hig__global-nav__top-nav__project-account-switcher__item__image-wrapper\\">
        <img class=\\"hig__global-nav__top-nav__project-account-switcher__item__image\\" src=\\"newImage.jpg\\" alt=\\"newLabel\\" width=\\"32\\" height=\\"32\\">
        <span class=\\"hig__global-nav__top-nav__project-account-switcher__item__image-placeholder\\">T</span>
    </span>
    <span class=\\"hig__global-nav__top-nav__project-account-switcher__item__label\\">testLabel</span>
</div><!--ACCOUNTS-->
    </div>
    <div class=\\"hig__global-nav__top-nav__project-account-switcher__lists__list\\">
        <!--PROJECTS-->
    </div>
</div><!--SLOT-->
        </div>
    </div>
</div><!--FLYOUT-->
</div><!--PROJECT_ACCOUNT_SWITCHER-->
  <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--right\\">
  <!--SHORTCUT-->
  </div>
  <!--PROFILE-->
</div><!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;
